### PR TITLE
Miscellaneous fixes

### PR DESCRIPTION
**Pull request by devnoname120**

### CODE REVIEW

**Comment by devnoname120** [comment]
@bvschaik I rebased my branch. Could you review it before doing further changes?

**Comment by bvschaik** [comment]
Sorry, didn't see your comment :slightly_frowning_face:
I will review before doing any more changes. Could you rebase again?

File: src/UI/BuildingMenu.c [hunk 1]
```diff
@@ -254,8 +254,6 @@ static void buttonMenuItem(int item)
 			case BUILDING_FORT:
 				menu.selectedSubmenu = 24;
 				break;
-            default:
```
Comment (bvschaik) [inline src/UI/BuildingMenu.c]: GCC 4.9 actually complains about switches on enums that don't include a default case, that's why it's here.

**Comment by devnoname120** [comment]
Sure

File: src/city/finance.c [hunk 2]
```diff
@@ -11,7 +11,7 @@ int city_finance_treasury()
 
 int city_finance_out_of_money()
 {
-    return Data_CityInfo.treasury <= MIN_TREASURY;
+    return Data_CityInfo.treasury <= (MIN_TREASURY);
```
Comment (bvschaik) [inline src/city/finance.c]: MIN_TREASURY is an integer constant, why use parens here?

File: src/Video/smacker.h [hunk 3]
```diff
@@ -39,7 +39,7 @@ typedef struct smk_t* smk;
 #define SMK_DONE	0x00
 #define SMK_MORE	0x01
 #define SMK_LAST	0x02
-#define SMK_ERROR	-1
+#define SMK_ERROR    (-1)
```
Comment (bvschaik) [inline src/Video/smacker.h]: This is code from an external library, I'd rather not change anything there.

File: src/UI/BuildingInfo_Military.c [hunk 4]
```diff
@@ -1,21 +1,19 @@
+#include <figure/formation.h>
```
Comment (bvschaik) [inline src/UI/BuildingInfo_Military.c]: Please use "" for local includes, and keep them together.

**Comment by devnoname120** [comment]
Done

**Comment by bvschaik** [comment]
Merged, thanks!

**Pull request merged**